### PR TITLE
Change `HtmlString` to inherit from `HtmlEncodedString`

### PR DESCRIPTION
**Pull request by dougbu**
- part of aspnet/HtmlAbstractions#5 fix

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTest.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ public class HtmlHelperTest
             {
                 return new TheoryData<string, string>
                 {
-                    { null, null },
+                    { null, string.Empty },
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTest.cs]: Was just strange that `HtmlString.ToString()` could previously return `null`.

**Comment by dougbu** [comment]
FYI slight improvement in memory performance with simple view that uses a view component that uses `AppendFormat()`. Most of the change comes from reducing the `StringWriter` count by one per request.
Before:
before
After:
after

**Comment by dougbu** [comment]
Note CI builds cannot succeed until aspnet/HtmlAbstractions#6 is in. Have tested combination locally of course.

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTest.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTest.cs]: agreed, very strange

**Comment by dougbu** [comment]
:up::date: per offline discussion.

@rynowak did you intend to sign off?

**Comment by rynowak** [comment]
:shipit: 

Ya sorry, I had intended to sign off yesterday, I got distracted by something distracting.

**Comment by dougbu** [comment]
3d82255

**Pull request closed without merging**